internal/bminventory/inventory_test.go

```diff
@@ -5401,6 +5401,181 @@ var _ = Describe("UpdateDiscoveryIgnition", func() {
+		addHost(clusterId, vsphereInventory, role)
+	}
+
+	addGenericHost := func(clusterId strfmt.UUID, role models.HostRole) {
+		const genericInventory = "{\"system_vendor\": {\"manufacturer\": \"Red Hat\", \"product_name\": \"KVM\", \"serial_number\": \"\", \"virtual\": true}}"
+		addHost(clusterId, genericInventory, role)
+	}
+
+	validateInventory := func(host models.Host, manufacturer string) bool {
+		var inventory models.Inventory
+		if err := json.Unmarshal([]byte(host.Inventory), &inventory); err != nil {
+			Expect(err).ShouldNot(HaveOccurred())
+		}
+		return inventory.SystemVendor.Manufacturer == manufacturer
+	}
+
+	validateHostsInventory := func(vsphereHostsCount int, genericHostsCount int) {
+		getReply := bm.GetCluster(ctx, installer.GetClusterParams{ClusterID: clusterID}).(*installer.GetClusterOK)
+		Expect(len(getReply.Payload.Hosts)).Should(Equal(vsphereHostsCount + genericHostsCount))
+		vsphereHosts := 0
+		genericHosts := 0
+		for _, h := range getReply.Payload.Hosts {
+			if validateInventory(*h, common.VmwareManufacturer) {
+				vsphereHosts++
+			}
+			if validateInventory(*h, "Red Hat") {
+				genericHosts++
+			}
+		}
+
+		Expect(vsphereHosts).Should(Equal(vsphereHostsCount))
+		Expect(genericHosts).Should(Equal(genericHostsCount))
+
+	}
+
+	getClusterPlatforms := func() *[]models.PlatformType {
+		platformReplay := bm.GetClusterSupportedPlatforms(ctx, installer.GetClusterSupportedPlatformsParams{ClusterID: clusterID})
+		Expect(platformReplay).Should(BeAssignableToTypeOf(installer.NewGetClusterSupportedPlatformsOK()))
+		return &platformReplay.(*installer.GetClusterSupportedPlatformsOK).Payload
+
+	}
+
+	It("no hosts", func() {
+		platformReplay := bm.GetClusterSupportedPlatforms(ctx, installer.GetClusterSupportedPlatformsParams{ClusterID: clusterID})
+		Expect(platformReplay).Should(BeAssignableToTypeOf(installer.NewGetClusterSupportedPlatformsOK()))
+		platforms := platformReplay.(*installer.GetClusterSupportedPlatformsOK).Payload
+		Expect(len(platforms)).Should(Equal(1))
+		Expect(platforms[0]).Should(Equal(models.PlatformTypeBaremetal))
+	})
+
+	It("single vsphere host", func() {
+		addVsphereHost(clusterID, models.HostRoleMaster)
+
+		validateHostsInventory(1, 0)
+		platformReplay := bm.GetClusterSupportedPlatforms(ctx, installer.GetClusterSupportedPlatformsParams{ClusterID: clusterID})
+		Expect(platformReplay).Should(BeAssignableToTypeOf(installer.NewGetClusterSupportedPlatformsOK()))
+		platforms := platformReplay.(*installer.GetClusterSupportedPlatformsOK).Payload
+		Expect(len(platforms)).Should(Equal(1))
+		Expect(platforms[0]).Should(Equal(models.PlatformTypeBaremetal))
+	})
+
+	It("3 vsphere hosts", func() {
+		supportedPlatforms := []models.PlatformType{models.PlatformTypeVsphere, models.PlatformTypeBaremetal}
+
+		addVsphereHost(clusterID, models.HostRoleMaster)
```

Comment:
is it possible to create a function which generate number of hosts? maybe use case use the builder design pattern here (it might be overkill)

Reply:
Not sure it's needed, you will need to receive the number of hosts and role for each one of them.
That is not really needed for my opinion (and add more complexity)